### DISCUSSION TITLE
Background curve movement questions...

### DISCUSSION THREAD

**Discussion by hermetheuscoffee** [artisan-roaster-scope/artisan#1026] [Q&A]
I have two questions related to moving a background curve in Artisan:

1. When moving a background curve (either by keyboard command or by the UI), is this logged anywhere in the roast? I don't see it captured in the "data" tab of a roast. It would be beneficial to see when a background curve was moved. (The reason is that when using PID, moving the background curve can cause wild fluctuations in heat... the operator of the roaster accuses the PID of going haywire by maxing out the heat, when MY theory is that the background profile was moved... causing the BT to suddenly appear far below the SV curve). 
2. I find that the right/left arrow keys sometimes control the position of the background curve, and at other times control one of my sliders. What toggles the behavior of the arrow keys? (I'd ideally like them to ALWAYS control the background curve). 

Thanks!

**Comment by MAKOMO** [comment]
> I have two questions related to moving a background curve in Artisan:
> 
> 1. When moving a background curve (either by keyboard command or by the UI), is this logged anywhere in the roast?

No

> I don't see it captured in the "data" tab of a roast. It would be beneficial to see when a background curve was moved. (The reason is that when using PID, moving the background curve can cause wild fluctuations in heat... the operator of the roaster accuses the PID of going haywire by maxing out the heat, when MY theory is that the background profile was moved... causing the BT to suddenly appear far below the SV curve).

Current we do not plan to add tracking of the background alignement.

> 2. I find that the right/left arrow keys sometimes control the position of the background curve, and at other times control one of my sliders. What toggles the behavior of the arrow keys? (I'd ideally like them to ALWAYS control the background curve).

As with all UIs the input goes to the UI widget under focus. If sliders are visible usually one slider widget is under focus. The slider under focus is marked with a darker groove. Focus can moved using the tab key. To move the focus to the profile canvas just hide the sliders using the View menu or press the key s. After making the sliders visible again you might need first to press tab to move focus to one of the sliders.

To improve the cursor key (and page up/down key) input for your situation we added flags to deactivate keyboard control for sliders (including the SV slider) as well as background move (by default keyboard control is on as it was in previous releases). You can find those additional flags in the Background dialog (menu `Roast >> Background`) and in the sliders tab (menu `Config >> Events, 3rd tab Sliders`) tagged "Keyboard Control". This addition will be part of the next release.
